File: projects/mcp-servers/php/src/Tools/GreetTool.php
<?php

namespace McpServer\Tools;

use MCP\Server\Tool\BaseTool;
use MCP\Server\Tool\Attribute\ToolParameter;

/**
 * GreetTool — returns a personalised greeting.
 *
 * Parameters:
 *   name (string, required) — the person or entity to greet.
 *   style (string, optional) — one of "friendly", "formal", "shout".
 */
class GreetTool extends BaseTool
{
    public function __construct()
    {
        parent::__construct(
            name: 'greet',
            description: 'Generate a personalised greeting message.',
            parameters: [
                new ToolParameter(
                    name: 'name',
                    type: 'string',
                    description: 'The name of the person or entity to greet',
                    required: true,
                ),
                new ToolParameter(
                    name: 'style',
                    type: 'string',
                    description: 'Greeting style: friendly, formal, or shout',
                    required: false,
                    default: 'friendly',
                ),
            ],
        );
    }

    /**
     * Execute the greet tool.
     *
     * @param  array<string, mixed>  $arguments
     * @return string
     */
    public function execute(array $arguments): string
    {
        $name  = $arguments['name'];
        $style = $arguments['style'] ?? 'friendly';

        return match ($style) {
            'formal'  => "Greetings, {$name}. It is a pleasure to make your acquaintance.",
            'shout'   => "HELLO {$name}! HOW ARE YOU TODAY?",
            default   => "Hey {$name}! Hope you're having a great day 👋",
        };
    }
}

File: projects/mcp-servers/csharp/Tools/GreetTool.cs
using ModelContextProtocol;
using ModelContextProtocol.Server;
using System.ComponentModel;

namespace CSharpMcpServer.Tools;

[McpTool("greet")]
[Description("Greets a user by name and returns a friendly message.")]
public sealed class GreetTool
{
    [McpToolParameter]
    [Description("The name of the person to greet.")]
    public string Name { get; set; } = string.Empty;

    [McpToolParameter]
    [Description("Optional title (e.g., Mr., Ms., Dr.).")]
    public string? Title { get; set; }

    [McpToolMethod]
    [Description("Generates a personalized greeting message.")]
    public string Execute()
    {
        var prefix = string.IsNullOrWhiteSpace(Title) ? string.Empty : $"{Title} ";
        return $"Hello, {prefix}{Name}! Welcome to C# MCP Server. The current time is {DateTime.Now:HH:mm:ss} UTC.";
    }
}
